app/main/views/g_cloud.py

```diff
@@ -265,3 +275,143 @@ def search_services():
         'search/services.html',
         **template_args
     )
+
+
+@direct_award.route('/save-search', methods=['GET'])
+def save_search():
+    # if there are multiple live g-cloud frameworks, we must assume the same filters work on them all
+    all_frameworks = data_api_client.find_frameworks().get('frameworks')
+    framework = framework_helpers.get_latest_live_framework(all_frameworks, 'g-cloud')
+    lots_by_slug = framework_helpers.get_lots_by_slug(framework)
+    current_lot_slug = get_lot_from_request(request, lots_by_slug)
+
+    # the bulk of the possible filter parameters are defined through the content loader. they're all boolean and the
+    # search api uses the same "question" labels as the content for its attributes, so we can really use those labels
+    # verbatim. It also means we can use their human-readable names as defined in the content
+    content_manifest = content_loader.get_manifest(framework['slug'], 'search_filters')
+    # filters - an OrderedDictionary of dicts describing each parameter group
+    filters = filters_for_lot(
+        current_lot_slug,
+        content_manifest,
+        all_lots=framework['lots']
+    )
+    clean_request_query_params = clean_request_args(request.args, filters.values(), lots_by_slug)
+
+    search_api_response = search_api_client.search_services(
+        index=framework['framework'],  # TODO: change this to framework['slug'], also search_presenters.py:~132
+        **build_search_query(request.args, filters.values(), content_manifest, lots_by_slug)
+    )
+    search_summary = SearchSummary(
+        search_api_response['meta']['total'],
+        clean_request_query_params.copy(),
+        filters.values(),
+        lots_by_slug
+    )
+
+    search_api_url = search_api_client.get_search_url(
+        index=framework['framework'],
+        **build_search_query(request.args, filters.values(), content_manifest, lots_by_slug)
+    )
+
+    return render_template(
+        'direct-award/save-search.html',
+        search_summary=search_summary,
+        search_api_url=search_api_url,
+    )
+
+
+@direct_award.route('/projects/create', methods=['POST'])
+def project_create():
+    try:
+        api_project = data_api_client.create_direct_award_project(user_id=current_user.id,
+                                                                  user_email=current_user.email_address,
+                                                                  project_name=request.form.get('project_name',
+                                                                                                'My new project'))
+
+    except HTTPError as e:
+        abort(e.status_code)
```

Comment:
We should probably pass on the message as well as the code in these cases?

Reply:
We generally haven't been doing this in other places in the app.